### PR TITLE
feat(CloudControl): Introduces list operations for bunch of services so cloudcontrol can pick them up

### PR DESCRIPTION
**Pull request by Pive01**
## Motivation
In order to support the [Resource Overview](https://github.com/localstack/localstack-web/pull/1459/files) project we need some:
- CloudControl apis to list resources
- Endpoint to gather current usage

## Changes
 Introduce CloudFormation list-types operations to list supported types by cloudcontrol

Relative -ext PR https://github.com/localstack/localstack-ext/pull/3699
/cc @simonrw @dominikschubert

### CODE REVIEW

**Comment by Pive01** [comment]
I have read the CLA Document and I hereby sign the CLA

File: localstack-core/localstack/services/apigateway/resource_providers/aws_apigateway_resource.py [hunk 1]
```diff
@@ -94,6 +96,30 @@ def read(
         """
         raise NotImplementedError
 
+    def list(
+        self,
+        request: ResourceRequest[ApiGatewayResourceProperties],
+    ) -> ProgressEvent[ApiGatewayResourceProperties]:
+        input_payload = json.loads(request.desired_state)
```
Comment (giograno) [inline localstack-core/localstack/services/apigateway/resource_providers/aws_apigateway_resource.py:103]: @simonrw I was playing with the resource endpoint and I came across this.
This is the only `list` operation so far that requires a desired state to be provided by `cloudcontrol`. When this does not happen, we pass an empty dictionary here, which of course fails to be JSON-loaded. From the rest of the code it looks like we could bound `Properties` to `dict`.

Comment (simonrw) [reply localstack-core/localstack/services/apigateway/resource_providers/aws_apigateway_resource.py]: What we should do to be in parity here is raise an exception if the required payload is not provided. I get the error message:

```
botocore.errorfactory.InvalidRequestException: An error occurred (InvalidRequestException) when calling the ListResources operation: Missing or invalid ResourceModel property in AWS::ApiGateway::Resource list handler request input.Required property:  (#: required key [RestApiId] not found)
```

I have just pushed a change that should make input handling better.

File: tests/integration/services/test_internal.py [hunk 2]
```diff
@@ -61,3 +61,18 @@ def test_get(self):
         assert doc["machine_id"]
         assert doc["system"]
         assert type(doc["is_license_activated"]) == bool
+
+
+@pytest.mark.usefixtures("openapi_validate")
```
Comment (giograno) [inline tests/integration/services/test_internal.py:77]: Kudos for adding a test 💯 !! This fixture won't have an effect till we define the endpoint spec in `openapi.yml`. I can take care of that in a second iteration!

**Review by simonrw — APPROVED** [review]

**Pull request merged**